From stable-bounces@linux.kernel.org  Tue Jul 26 16:40:13 2005
Date: Tue, 26 Jul 2005 16:40:31 -0700 (PDT)
To: stable@kernel.org
From: "David S. Miller" <davem@davemloft.net>
Subject: [PATCH][XFRM]: Fix possible overflow of sock->sk_policy

From: Herbert Xu <herbert@gondor.apana.org.au>

[XFRM]: Fix possible overflow of sock->sk_policy

Spotted by, and original patch by, Balazs Scheidler.

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Chris Wright <chrisw@osdl.org>
---

 net/xfrm/xfrm_user.c |    3 +++
 1 files changed, 3 insertions(+)

Index: linux-2.6.12.y/net/xfrm/xfrm_user.c
===================================================================
--- linux-2.6.12.y.orig/net/xfrm/xfrm_user.c
+++ linux-2.6.12.y/net/xfrm/xfrm_user.c
@@ -1180,6 +1180,9 @@ static struct xfrm_policy *xfrm_compile_
 	if (nr > XFRM_MAX_DEPTH)
 		return NULL;
 
+	if (p->dir > XFRM_POLICY_OUT)
+		return NULL;
+
 	xp = xfrm_policy_alloc(GFP_KERNEL);
 	if (xp == NULL) {
 		*dir = -ENOBUFS;
